include/dfu/dfu_target.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef DFU_TARGET_H__
+#define DFU_TARGET_H__
+
+/**
+ * @defgroup dfu_target DFU Target 
+ * @brief DFU Target 
+ *
+ * @{
+ */
+
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Functions which needs to be supported by all DFU targets.
+ */
+struct dfu_target {
```

Comment:
even identify?  I mean it can't really return anything except true/false but if you want it like the socket api where it can only return `0`/`-1` I'm not against it. It would just feel a bit weird.

Reply:
If it can't fail it's okay to just return `bool`.

Reply:
left identify as is. It's a simple check function which can't really fail.